### PR TITLE
CF: Mapping Caution Flag Content: Database changes

### PR DESCRIPTION
**Pull request by zurbergram**
## Description
Main PR: https://github.com/department-of-veterans-affairs/gibct-data-service/pull/630

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by shawkey-daniel — APPROVED** [review]
LGTM

File: db/migrate/20200327142848_create_caution_flag_rules.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      t.timestamps
+    end
+
+    add_foreign_key :caution_flag_rules, :rules, validate: false
+  end
+
+  def down
+    drop_table :caution_flag_rules
+
+    remove_foreign_key :caution_flag_rules, :rules
```
Comment (jholton) [inline db/migrate/20200327142848_create_caution_flag_rules.rb:18]: If you try to rollback you get an error.  If you're dropping the table, you don't need to also manually remove the foreign key.

File: db/migrate/20200327142848_create_caution_flag_rules.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+class CreateCautionFlagRules < ActiveRecord::Migration[5.2]
+  def up
+    create_table :caution_flag_rules do |t|
+      t.integer :rule_id
```
Comment (jholton) [inline db/migrate/20200327142848_create_caution_flag_rules.rb:4]: Rails makes these foreign keys `bigint`s now.  And this doesn't get you an index on `rule_id`.  Try this instead:
```ruby
    create_table :caution_flag_rules do |t|
      t.belongs_to :rule, foreign_key: true
      t.string :title
      t.string :description
      t.string :link_text
      t.string :link_url
      t.timestamps
    end
```
and you don't need the `add_foreign_key` below either.

Comment (zurbergram) [reply db/migrate/20200327142848_create_caution_flag_rules.rb]: changed

File: db/migrate/20200327142848_create_caution_flag_rules.rb (hunk 1 above)
Comment (zurbergram) [reply db/migrate/20200327142848_create_caution_flag_rules.rb]: oh cause the order is wrong, removed manually removing the foreign key

**Review by jholton — APPROVED** [review]

**Pull request merged**